src/solver/optimisation/base_weekly_optimization.cpp

```diff
@@ -41,18 +43,30 @@ WeeklyOptimization::WeeklyOptimization(const OptimizationOptions& options,
 {
 }
 
-std::unique_ptr<WeeklyOptimization> WeeklyOptimization::create(
-    const Antares::Data::Study& study,
-    const OptimizationOptions& options,
-    AdqPatchParams& adqPatchParams,
-    PROBLEME_HEBDO* problemeHebdo,
-    uint thread_number,
-    IResultWriter& writer)
+std::unique_ptr<WeeklyOptimization> WeeklyOptimization::create(const Antares::Data::Study& study,
```

Comment:
**create** function : good example where return type could be moved up, as it is quite large

Reply:
Except if we prefer to **always** have return type on the previous line, this will not be possible here, because it will anyway be necessary to have each argument on its line.
So it will not be needed to have the return type on its own line, because the line will already be under 100 chars.